src/home/homework-luks.c

```diff
@@ -1901,27 +1903,31 @@ static int make_partition_table(
         if (r < 0)
                 return log_error_errno(r, "Failed to place partition at first free partition index: %m");
 
+        // Use same sector size as the fdisk context when converting to bytes
+        fdisk_sector_size = fdisk_get_sector_size(c);
+        assert(fdisk_sector_size);
```

Comment:
coding style: for arithmetic values we prefer explicit non-null comparisons. only for bools (and boolean-like values) and pointers we are fine with using C's "downgrade-to-bool" syntax.

or in other words:

`assert(fdisk_sector_size > 0);`

here please